corporate/lib/stripe.py

```diff
@@ -4014,6 +4083,16 @@ def get_upgrade_page_session_type_specific_context(
             is_self_hosting=True,
         )
 
+    @override
+    def check_valid_plan_tier(self, plan_tier: int) -> bool:  # nocoverage
+        valid_tiers = [
+            CustomerPlan.TIER_SELF_HOSTED_BASIC,
+            CustomerPlan.TIER_SELF_HOSTED_BUSINESS,
+        ]
+        if plan_tier in valid_tiers:
+            return True
+        return False
```

Comment:
I'm not sure I love this function name, given that there are other valid plan tiers -- even if not valid to transition to in some cases.

Reply:
Changed from `check_valid_plan_tier` to `check_plan_tier_is_billable`. Another thought I had was `check_plan_tier_is_implemented`? It's mostly about whether the plan tier is appropriate for the billing entity (e.g. we don't want to accidentally set a Zulip Cloud tier on a self-hosted billing entity).

Reply:
This is probably fine, though maybe worth a discussion in Zulip about further refactoring around tier definitions to make things nice and readable.